src/components/Icon/Icon.style.ts

```diff
@@ -1,20 +1,35 @@
 import { rem } from 'polished';
 import { Theme } from '../../theme';
 import { AcceptedColorComponentTypes, fillPickerBasedOnType } from '../../utils/themeFunctions';
+import { pickTextColorFromSwatches } from '../../theme/palette';
+import { ColorConfig } from '../TextField/TextField.style';
 
 type iconStyleProps = {
   /** Property indicating the color of the icon. Defaults to primary */
   color: AcceptedColorComponentTypes | string;
   /** Property indicating the size of the icon. Defaults to 16 */
   size: number;
+  colorConfig?: ColorConfig;
 };
 
-export const iconStyle = ({ color, size }: iconStyleProps) => (theme: Theme) => ({
-  fill: fillPickerBasedOnType(color)(theme),
+const getIconColor = (
+  color: AcceptedColorComponentTypes | string,
+  theme: Theme,
+  colorConfig?: ColorConfig
```

Comment:
i dont feel colorConfig needed here at all as if the parent has this info you can easily pass the color string to the icon component. Icon component is from the few components that can get any color

Reply:
Hmm, did I miss something in the structure of our DS's palette? Since this is a tricky one, we can take it offline. 

Just a brief overview of the colorConfig usage:

We declare a 'leftIcon'  in our TextField component
```
      <TextField
        fill={..}
        fillShade={...}
        leftIcon={
          <Icon name={select('name',Object.keys(iconSelector),'search')} />}
       />
```

Then we want to change the Icon's color based on the background color of the textField.

in TextField.jsx
```
<div>
  ....
// Issue
     {leftIcon && leftIcon } // Here we need a functional component here to pass our current TextField fill/fillshade not a JSX Element

//So we have to use the leftIcon like a functional component to pass the textField props inside it

{leftIcon && leftIcon(fill, fillShade)}
</div>
```
That said leftIcon should have the following structure based on the render-props pattern
```
const leftIcon = (fill, fillShde) =>  <Icon fill={fill} fillShade={fillShade} />
```

Is this explanation clear?

Reply:
use the icon as an example on the usage as well like the 

![image](https://user-images.githubusercontent.com/6433679/102232127-6955b180-3ef7-11eb-93b8-9a4156b79841.png)